salt/utils/xmlutil.py

```diff
@@ -300,7 +300,7 @@ def change_xml(doc, data, mapping):
                 if convert_fn:
                     new_value = convert_fn(new_value)
 
-                if current_value != new_value:
+                if str(current_value) != str(new_value):
```

Comment:
This should be moved into its own commit to ease patches backporting if needed in the future.

Reply:
roger that